### ISSUE TITLE
Is it possibile to have 2-dimensional lists as genes type?

### ISSUE THREAD

**Issue by DarkFighterLuke** [ahmedfgad/GeneticAlgorithmPython#110]
Well, the title is pretty straightforward.
My population is a set of 2-dimensional lists filled with boolean values, each one representing a solution, but i get `ValueError: A 2D list is expected to the initail_population parameter but a (3-D) list found.` when setting `initial_population` and I have found no information on how (if possible) to set `gene_type` to 2-dimensional lists.

**Comment by ahmedfgad** [comment]
Unfortunately, this is not supported in PyGAD yet.

The `initail_population` parameter must be a 2D list where a single row represents a single solution. Also, the value passed to `gene_type` cannot be a 2D list.

Things are not straightforward to make these changes because you have to also change how the crossover and mutation work.

**Comment by DarkFighterLuke** [comment]
Thank you for your answer. Finally I managed to get it work and I've exactly done what you just said.

Just for the sake of telling, in case someone else will face this same issue:
I mapped each 2D chromosome that I had to a unique id, so basically I got a population like this:
`[ [0], [1], [2], ... ]`
(Obviously the mapping is stored in a global variable, so that it can be accessed from fitness, crossover and mutation functions as it follows)

At this point, I defined custom fitness, crossover and mutation functions, which get back the original 2D chromosome according to the id received and work on it. At the end, in the case of the crossover function, the newly created chromosome is assigned a unique id and saved to the structure storing the mapping (it can be memory inefficient, but that wasn't my case).

Source code can be found [here](https://github.com/DarkFighterLuke/GAEnergyOptimizer).
